esp/services/ws_workunits/ws_workunitsHelpers.cpp

```diff
@@ -898,6 +898,38 @@ void WsWuInfo::getGraphTimingData(IArrayOf<IConstECLTimingData> &timingData)
         it->playProperties(visitor);
 }
 
+void WsWuInfo::getSOAPDatasetNodes(IEspECLWorkunit &info, unsigned long flags)
+{
+    if (!(flags & WUINFO_IncludeSOAPDatasetNode))
+        return;
+
+    IArrayOf<IConstSOAPDatasetNode> datasetNodes;
+    Owned<IConstWUGraphIterator> graphs = &cw->getGraphs(GraphTypeActivities);
+    ForEach(*graphs)
+    {
+        Owned <IPropertyTree> xgmml = graphs->query().getXGMMLTree(false);
+        Owned<IPropertyTreeIterator> iter = xgmml->getElements("//node[@label='SOAP dataset']");
```

Comment:
@wangkx 
Technically the label is a description and could change, the real identifier is the ` <att name="_kind" value="55"/>` 
Where value is a few different possibilities.
But I realize that filtering by an att named value may be very slow.
If you can't filter by kind, the label should be stable at least for now.

Currently the label can be several values which also complicates your filtering.
"SOAP dataset", "SOAP action", "HTTP dataset"

Reply:
Could we assume that the WUInfo user knows the label and add the label to WUInfoRequest?

Reply:
I would have thought it should be by activity kind ..
But then should this service be agnostic about what it's been asked to include, i.e. why just a particular set of kinds?
Should the service cope with a list of activity types to find? So e.g. could request CSV Write activities instead.

I'm a bit concerned about the efficiency and requirements here.
The result will be a flattened (i.e. you will no longer be able to tell which graph/subgraph they came from) list of activities found, with all their attributes.
Ultimately how is the info. consumed by the client requesting it?
@GordonSmith @wangkx

Reply:
@jakesmith it's unfortunate that kind is an attribute of node.  Fits in with a discussion of how inefficient xgmml can be.
Is there an efficient way of filtering nodes that fall into a set of activity kinds?  SOAP and HTTP call alone have a total of 5 kinds.
I don't think this use case is about understanding the graph of the query.  It's more about knowing what services etc the query calls out to.  If the list of soap and http calls were cached somewhere more accessible that would work to.